src/main/java/core/basesyntax/SalaryInfo.java

```
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private LocalDate parsedDateFrom;
+    private LocalDate parsedDateTo;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder resultInfo = new StringBuilder();
+
+        // reading and parsing Strings dateFrom, dateTo
+        try {
+            parsedDateFrom = LocalDate.parse(dateFrom, dateFormatter);
+            parsedDateTo = LocalDate.parse(dateTo, dateFormatter);
+            resultInfo.append("Report for period ");
+            resultInfo.append(dateFrom).append(" - ").append(dateTo)
+                    .append(System.lineSeparator());
+        } catch (NullPointerException | DateTimeException e) {
+            System.out.println("Date should be not NULL! and be in \"dd.MM.yyyy\" format");
+            System.out.println(e.getMessage());
```

The `DateTimeException` is thrown when the date format is invalid. In this case, you should throw the exception instead of catching it to indicate that the input is invalid. This aligns better with the program's responsibility of validating input data.